src/crimson/osd/heartbeat.h

```diff
@@ -166,33 +172,55 @@ class Heartbeat::Peer {
     }
   }
 
-  void connect();
-  void disconnect();
+  // a session starts when con_front and con_back are both connected
+  void start_session();
+  // a session resets when either con_front or con_back is reset
+  void reset_session();
+  // notify when con_front becomes ready, possibly start session
+  void notify_front_ready();
+  void connect_front();
+  // notify when con_back becomes ready, possibly start session
+  void notify_back_ready();
+  void connect_back();
+
+  void do_send_heartbeat(clock::time_point now,
+                         ceph::signedspan mnow,
+                         std::vector<seastar::future<>>*);
 
  private:
   Heartbeat& heartbeat;
   const osd_id_t peer;
 
-  /// peer connection (front)
-  crimson::net::ConnectionRef con_front;
-  /// peer connection (back)
-  crimson::net::ConnectionRef con_back;
-  /// time we sent our first ping request
+  // time we sent our first ping request
   clock::time_point first_tx;
-  /// last time we sent a ping request
+  // last time we sent a ping request
```

Comment:
please note `///` is not a typo, the idea is to let doxygen to reader this comment as the document of the following member variable.

Reply:
noted, do you think it is necessary to be rendered by doxygen?

Reply:
@cyx1231st i don't intent to argue that if a comment will be rendered by doxygen or not . we have doyxgen style comment everywhere in our source tree. i don't think it's wrong or inappropriate . and i don't see this change helps to improve the readability of runtime efficiency in any way.